cnf-tests/testsuites/e2esuite/nmstate/nmstate_sriov.go

```diff
@@ -0,0 +1,200 @@
+
+	BeforeAll(func() {
+		if !networks.IsSriovOperatorInstalled() {
+			Skip("SR-IOV operator not installed on the cluster.")
+		}
+
+		err := namespaces.Create(testNamespace, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("CleanSriov...")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovInfos, err = cluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(sriovInfos.Nodes)).ToNot(BeZero(), "no node with SR-IOV NICs found")
+
+		nmstate := nmstatev1.NMState{ObjectMeta: metav1.ObjectMeta{Name: "nmstate"}}
+		err = client.Client.Create(context.Background(), &nmstate)
+		Expect(err).ToNot(HaveOccurred())
+		DeferCleanup(client.Client.Delete, context.Background(), &nmstate)
+	})
+
+	Context("when a SriovNetworkNodePolicy and NodeNetworkConfigurationPolicySpec target the same NIC", func() {
+
+		BeforeAll(func() {
+			namespaces.CleanPods(testNamespace, client.Client)
+		})
+
+		It("VFs should spawn correctly", func() {
+			node, devices, err := findNonPrimarySriovDevicesAndNode(sriovInfos)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(devices)).ToNot(BeZero())
+
+			By("Using node " + node)
+
+			testDevice := devices[0]
+			By("Using device " + testDevice.Name)
+
+			By("Configuring IPAM on Physical Function via NMState")
+			nodeNetworkConfigPolicy := nmstatev1.NodeNetworkConfigurationPolicy{
+				ObjectMeta: metav1.ObjectMeta{Name: "test-sriov-device-" + testDevice.Name},
+				Spec: nmstateshared.NodeNetworkConfigurationPolicySpec{
+					NodeSelector: map[string]string{
+						"kubernetes.io/hostname": node,
+					},
+					DesiredState: nmstateshared.State{
+						Raw: nmstateshared.RawState(fmt.Sprintf(nmstateDesiredStateTemplateStaticIP, testDevice.Name)),
+					},
+				},
+			}
+			err = client.Client.Create(context.Background(), &nodeNetworkConfigPolicy)
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(client.Client.Delete, context.Background(), &nodeNetworkConfigPolicy)
+
+			waitForNMStatePolicyToBeStable(&nodeNetworkConfigPolicy)
+
+			By("Configuring VF on device via SriovNetworkNodePolicy")
+			sriovNetworkNodePolicy, err := sriovnetwork.CreateSriovPolicy(sriovclient, "test-nmstate-", namespaces.SRIOVOperator, testDevice.Name, node, 10, "testnmstateresource", "netdevice")
+			Expect(err).ToNot(HaveOccurred())
+			DeferCleanup(sriovclient.Delete, context.Background(), sriovNetworkNodePolicy)
+
+			Eventually(func() int64 {
+				testedNode, err := sriovclient.Nodes().Get(context.Background(), node, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				resNum := testedNode.Status.Allocatable[corev1.ResourceName("openshift.io/testnmstateresource")]
+				capacity, _ := resNum.AsInt64()
+				return capacity
+			}, 10*time.Minute, time.Second).Should(Equal(int64(10)))
+		})
+	})
+})
+
+func findNonPrimarySriovDevicesAndNode(n *cluster.EnabledNodes) (string, []*sriovv1.InterfaceExt, error) {
+	errs := []error{}
```

Comment:
You could use this lib  https://pkg.go.dev/errors  to wrap errors

Reply:
I'm using that package at the end of the function:
```
if len(retDevices) == 0 {
	return "", nil, fmt.Errorf("can't find any SR-IOV devices in cluster's nodes: %w", errors.Join(errs...))
}
```

How would you use it with `Wrap(...)`?

Reply:
I think you can avoid the []array
```
func main() {
	var err error
    if false{
	err1 := errors.New("err1")
	err = errors.Join(err, err1)
	err2 := errors.New("err2")
	err = errors.Join(err, err2)
}
	fmt.Println(err)

}
```
but to be honest, both works